Kernel/Syscalls/sched.cpp

```diff
@@ -27,6 +27,9 @@ ErrorOr<NonnullRefPtr<Thread>> Process::get_thread_from_pid_or_tid(pid_t pid_or_
         peer = Thread::current();
         if (pid_or_tid != 0)
             peer = Thread::from_tid(pid_or_tid);
+        // Only superuser can access other processes' threads.
+        if (!credentials()->is_superuser() && peer && &peer->process() != this)
+            return EPERM;
```

Comment:
This could probably be squashed into the commit that added `get_thread_from_pid_or_tid` in the first place. That said, I'm not sure if `Process::get_thread_from_pid_or_tid` should be doing credential checks at all, as it sounds like that may very well be dependent on what we are using the returned thread for.

Reply:
I do actually think that unless we're doing some very low level stuff anything the user asks for should be protected in this way. I'd leave it until we actually have another use case, because extracting this check leads to a bit of extra code (we need to re-find the pid).